irc/server.py

```diff
@@ -155,7 +156,7 @@ def _handle_incoming(self):
 
         self.buffer.feed(data)
         for line in self.buffer:
-            line = line.decode('utf-8')
+            line = line.decode(self.incoming_encoding)
```

Comment:
Here I prefer to keep the hard-coded value. I'm not aware of any use case that would demand any other encoding.